### PR TITLE
first draft

### PR DESCRIPTION
**Pull request by markedwardmurray**
Problems 9-12 may be difficult in their current form. I could make them simpler or tag them as "advanced".

### CODE REVIEW

File: Collections/FISAppDelegate.m [hunk 1]
```diff
@@ -10,40 +10,193 @@
...
+    [mutableArray exchangeObjectAtIndex:0 withObjectAtIndex:endIndex];
+    return [NSArray arrayWithArray:mutableArray];
+}
+
+-(NSArray *)arrayByReversingArray:(NSArray *)array {
+    NSMutableArray *mutableArray = [array mutableCopy];
+    NSUInteger i = 0;
+    NSUInteger j = mutableArray.count - 1;
+    
+    for ( ; i < (mutableArray.count / 2); i++) {
```
Comment (jmburges) [inline Collections/FISAppDelegate.m]: woah. Why are you doing the `i=0` outside of the for loop?

File: Collections/FISAppDelegate.m [hunk 2]
```diff
@@ -10,40 +10,193 @@
...
+
+    return [NSArray arrayWithArray:hobbits];
 }
 
-- (void)applicationWillEnterForeground:(UIApplication *)application
-{
-    // Called as part of the transition from the background to the inactive state; here you can undo many of the changes made on entering the background.
+-(NSArray *)arrayByFilteringStringsBeginningWithA:(NSArray *)array {
+    
+    NSPredicate *beginsWithAPredicate = [NSPredicate predicateWithFormat:@"self BEGINSWITH[c] 'a'"];
```
Comment (jmburges) [inline Collections/FISAppDelegate.m:103]: did you teach `cd` as the way to do do case-insensitive compare stuff?

Comment (markedwardmurray) [inline Collections/FISAppDelegate.m:103]: yes, it's always been in the reading on NSPredicate.

File: Collections/FISAppDelegate.m [hunk 3]
```diff
@@ -10,40 +10,193 @@
...
+    
+    NSString *lowercaseNoPunctuation = [noPunctuation lowercaseString];
+    
+    NSArray *arrayOfWords = [lowercaseNoPunctuation componentsSeparatedByString:@" "];
+    
+    NSMutableDictionary *countsOfWords = [[NSMutableDictionary alloc] init];
+    
+    for (NSString *word in arrayOfWords) {
+        if ([countsOfWords[word] integerValue] > 0) {
+            NSInteger count = [countsOfWords[word] integerValue];
```
Comment (jmburges) [inline Collections/FISAppDelegate.m]: I would do the count increment in one line. So just `[countsOfWords[word] integerValue] +1`. Keeps everything a bit more together. I was looking around for why you were incrementing it, thinking it was a like loop variable or something

File: Collections/FISAppDelegate.m (hunk 1 above)
Comment (markedwardmurray) [inline Collections/FISAppDelegate.m]: I wasn't sure how to declare both NSUInteger `i` and `j` within the statement and i wanted to keep them together. I just realized though that I don't think I need `j` at all.

File: Collections/FISAppDelegate.m (hunk 2 above)
Comment (jmburges) [inline Collections/FISAppDelegate.m:103]: so then why did you use `c` here instead of `cd`?

File: Collections/FISAppDelegate.m (hunk 1 above)
Comment (jmburges) [inline Collections/FISAppDelegate.m]: I would put the i in the looping stuff. The j can be separte but that i=0 wants to be in the for declaration

File: Collections/FISAppDelegate.m (hunk 2 above)
Comment (markedwardmurray) [inline Collections/FISAppDelegate.m:103]: there aren't any diactritics in the test strings, so I just used `[c]`

File: Collections/FISAppDelegate.m (hunk 3 above)
Comment (markedwardmurray) [inline Collections/FISAppDelegate.m]: haha, i was actually trying to improve readability by doing that. oh well, i'll do it on one line.

File: Collections/FISAppDelegate.m (hunk 1 above)
Comment (markedwardmurray) [inline Collections/FISAppDelegate.m]: :+1:

**Pull request merged**